Model/RatepayInvoicePaymentMethod.php

```diff
@@ -0,0 +1,131 @@
+            $this->_oPayment->oxpayments__wdoxidee_maid->value,
+            $this->_oPayment->oxpayments__wdoxidee_secret->value
+        );
+
+        $oConfig->add($oPaymentMethodConfig);
+        return $oConfig;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return RatepayInvoiceTransaction
+     *
+     * @since 1.2.0
+     */
+    public function getTransaction()
+    {
+        return new RatepayInvoiceTransaction();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getConfigFields()
+    {
+        $aAdditionalFields = [
+            'descriptor' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_descriptor',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_descriptor'),
+                'description' => Helper::translate('wd_config_descriptor_desc'),
+            ],
+            'additionalInfo' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_additional_info',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_additional_info'),
+                'description' => Helper::translate('wd_config_additional_info_desc'),
+            ],
+            'deleteCanceledOrder' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_delete_canceled_order',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_delete_cancel_order'),
+                'description' => Helper::translate('wd_config_delete_cancel_order_desc'),
+            ],
+            'deleteFailedOrder' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_delete_failed_order',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_delete_failure_order'),
+                'description' => Helper::translate('wd_config_delete_failure_order_desc'),
+            ],
+        ];
+        return parent::getConfigFields() + $aAdditionalFields;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getPublicFieldNames()
+    {
+        return array_merge(
+            parent::getPublicFieldNames(),
+            ['additionalInfo', 'deleteCanceledOrder', 'deleteFailedOrder']
```

Comment:
is there a reason, why we do not have descriptor in here?

Reply:
There is not, added it 👍